toolbox/swiftsim/particle_splitting/generate_splitting_information.py

```diff
@@ -438,18 +572,20 @@ def gather_by_progenitor_id(data):
     order = psort.parallel_sort(target_task, return_index=True, comm=comm)
     for key, value in data.items():
         data[key] = psort.fetch_elements(value, order, comm=comm)
+    root_ids = psort.fetch_elements(root_ids, order, comm=comm)
```

Comment:
Something I do not understand about this is why we use root_ids as a separate array. Wouldn't it make more sense to overwrite the data['progenitor_ids'] with the root_ids? That would leave the code as it was before.

Reply:
We can't overwrite at this point since this is before we have assigned particles to ranks, and MPI cannot pass arrays with dtype 'object'. I added some extra comments to the code in https://github.com/SWIFTSIM/HBTplus/pull/61/commits/046944ff50720e402bc1a9b4f2df6a83ec5ba9e4